Source/drlg_l1.cpp

```diff
@@ -741,18 +741,17 @@ void ApplyShadowsPatterns()
 	}
 }
 
-bool PlaceMiniSet(const Miniset &miniset, bool setview)
+bool PlaceMiniSet(const Miniset &miniset, Point *position = nullptr)
 {
 	int sw = miniset.size.width;
 	int sh = miniset.size.height;
-	int sx = GenerateRnd(DMAXX - sw) - 1;
+	int sx = GenerateRnd(DMAXX - sw);
 	int sy = GenerateRnd(DMAXY - sh);
 
-	for (int bailcnt = 0;; bailcnt++) {
+	for (int bailcnt = 0;; bailcnt++, sx++) {

```

Comment:
I was thinking this could operate over a range where the iterator operates something like:

```cpp
class DRLGIterator {
  Point pos;
  unsigned iterationCount;
  operator++() {
    iterationCount++;
    x++;
    if (x > MaxX) {
      x = 0;
      y++;
      if (y > MaxY) y = 0;
    }
  }
}

DRLGIterator begin() { return { GenerateRnd(MaxX), GenerateRnd(MaxY), 0 }; }

DRLGIterator end() { return { 0, 0, 4000}; }
```

Keeping the increment logic together makes it easier to refactor to that end.

Reply:
It would be nice to get rid of the part that walks over the tiles, but I think I will leave such a change for a later PR, maybe someone else :)

I have unified the 4 versions of `PlaceMiniSet()` as part of this PR, which was the goal with this series, so it should at least be pretty easy to do later.